### PR TITLE
8306329 : Update ICU4C to 73.1

### PR DESCRIPTION
**Pull request by HimaBinduMeda**
Updated ICU to v73.1. Verified build and sanity. No issues seen.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8306329](https://bugs.openjdk.org/browse/JDK-8306329): Update ICU4C to 73.1

### Reviewers
 * [Kevin Rushforth](https://openjdk.org/census#kcr) (@kevinrushforth - **Reviewer**)
 * [Johan Vos](https://openjdk.org/census#jvos) (@johanvos - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx.git pull/1138/head:pull/1138` \
`$ git checkout pull/1138`

Update a local copy of the PR: \
`$ git checkout pull/1138` \
`$ git pull https://git.openjdk.org/jfx.git pull/1138/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1138`

View PR using the GUI difftool: \
`$ git pr show -t 1138`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1138.diff">https://git.openjdk.org/jfx/pull/1138.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jfx/pull/1138#issuecomment-1548913034)

### CODE REVIEW

File: modules/javafx.web/src/main/native/Source/ThirdParty/icu/LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 COPYRIGHT AND PERMISSION NOTICE
 
-Copyright © 1991-2022 Unicode, Inc. All rights reserved.
+Copyright © 1991-2023 Unicode, Inc. All rights reserved.
```
Comment (kevinrushforth) [inline modules/javafx.web/src/main/native/Source/ThirdParty/icu/LICENSE:3]: This LICENSE file is missing the same preamble you recently added to `icu_web.md` lines 6-18. Or you can copy from:

https://github.com/unicode-org/icu/blob/release-73-1/icu4c/LICENSE

**Review by kevinrushforth — COMMENTED** [review]
This looks good with one comment about a missing LICENSE section.

File: modules/javafx.web/src/main/native/Source/ThirdParty/icu/LICENSE (hunk 1 above)
Comment (HimaBinduMeda) [reply modules/javafx.web/src/main/native/Source/ThirdParty/icu/LICENSE:16]: Done

**Review by kevinrushforth — APPROVED** [review]

**Comment by kevinrushforth** [comment]
@johanvos @tiainen Can one of you be the second reviewer?

**Review by johanvos — APPROVED** [review]
Buld + tests ok

**Pull request closed without merging**